lib/documents.js

```diff
@@ -166,6 +166,12 @@ export async function register({
         if(upsertEntityResult instanceof Error) {
           throw upsertEntityResult;
         }
+        if(!upsertEntityResult) {
```

Comment:
Hmm, this would have a significantly negative performance impact on what may be the hottest path. Let me look around and see if there's a better way to deal with this.

Reply:
Ok. The situation this happens in is when there is no internalId, so `upsertEntityPromise` never gets set up to run.

Reply:
Yeah, `internalId` should always be passed in an optimized system. I've done what I can here to optimize and left a comment about how optimized systems should fetch the `internalId` first so it can be passed in the call.